Site management panel: Combine log tabs

fushar opened the pull request:
Related to:

- https://github.com/Automattic/dotcom-forge/issues/7514

## Proposed Changes

This PR combines the `PHP Logs` and `Server Logs` tabs into one single Logs tab:

|Before|After|
|-|-|
|PHP<br><img width="1357" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1525580/92fbaf6e-6076-4b1c-bc54-17ce69ba9ab1"><br><br>Web<br><img width="1357" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1525580/51795977-5ef0-4d8e-8580-5dd13b0a778c">|PHP<br><img width="1356" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1525580/a57959a8-8072-4cac-a4a8-321749c5fba2"><br><br>Web<br><img width="1356" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1525580/998e71fe-d703-4785-b165-889fe2dcb7a3">|

> [!NOTE]
> I am reusing the old `/site-logs/` route, which is now unused. Seems perfect for this job.

> [!NOTE]
> Here, I improvise the subtab naming based on my suggestion in the [issue](https://github.com/Automattic/dotcom-forge/issues/7514). Feel free to suggest otherwise, or we can iterate later.

## Why are these changes being made?

We want to refine the information architecture of the site management panel.

## Testing Instructions

1. Go to `/sites`.
2. Select an atomic site.
3. Click Logs tab.
4. Play around with the PHP subtab; verify that the functionality works.
5. Click the HTTP request subtab.
6. Play around with the HTTP request subtab; verify that the functionality works.

Regression testing

1. Visit `/site-monitoring/:site/php?flags=-layout/dotcom-nav-redesign-v2` and verify that it still works.
1. Visit `/site-monitoring/:site/web?flags=-layout/dotcom-nav-redesign-v2` and verify that it still works.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

fushar committed:
Site management panel: Combine log tabs

miksansegundo commented on client/site-logs/components/logs-header/index.tsx:
```diff
@@ -0,0 +1,57 @@
+				) }
+			/>
+			<div className="logs-header__selector-container">
+				<div className="logs-header__selector-heading">{ translate( 'Log type' ) }</div>
```
```suggestion
				<div className="logs-header__selector-heading">{ translate( 'Log Type' ) }</div>
```

To be consistent with:
<img width="273" alt="Screenshot 2567-05-31 at 15 12 30" src="https://github.com/Automattic/wp-calypso/assets/1881481/c65ac362-152d-417a-99d1-9fc4004a8de1">

fushar replied:
Thanks for noticing. See the latest comment 😄

taipeicoder reviewed (approved):
Works nicely, with two non-blocking notes:

1. There's a lot of repetition of the word "Logs" here. 
![Screenshot 2024-05-31 at 3 59 51 PM](https://github.com/Automattic/wp-calypso/assets/797888/19149c26-02e6-4ad4-be33-f6d600342c0a)

My proposal:
![Screenshot 2024-05-31 at 4 07 39 PM](https://github.com/Automattic/wp-calypso/assets/797888/f0184347-1366-4b48-a4ca-6d02dd9b72a7)

What's changed:
 - "Log type" to "View" as per Lucas' comment https://github.com/Automattic/dotcom-forge/issues/7514#issuecomment-2138022202.
 - "HTTP request logs" to "Webserver" as per the support docs https://developer.wordpress.com/docs/troubleshooting/site-monitoring/#php-logs-and-webserver-logs

2. If I am in Logs > HTTP request logs of Site A, and then click on Site B from the collapsed Sites list, I land in Logs > PHP errors, but I expected to land on Logs > HTTP request logs of Site B. This is a bit odd since the filters in Logs carry over from Site A to Site B, so I expected the selected tab to carry over as well.

We can address the latter in a follow-up PR, if needed 🙂

miksansegundo reviewed (approved):
LGTM 🚀 

<img width="1243" alt="Screenshot 2567-05-31 at 15 11 13" src="https://github.com/Automattic/wp-calypso/assets/1881481/34ba0ffc-7cf7-4366-8d42-abd703ef8ace">


<img width="1240" alt="Screenshot 2567-05-31 at 15 11 23" src="https://github.com/Automattic/wp-calypso/assets/1881481/df574933-acae-419d-b59e-0cc230cbb4f5">

miksansegundo commented:
> This is a bit odd since the filters in Logs carry over from Site A to Site B, so I expected the selected tab to carry over as well.
+1 to persist the state of the filters

Here is the recording of that issue

https://github.com/Automattic/wp-calypso/assets/1881481/57987cb2-7c04-46b5-8813-fafa9aac5bd6

fushar committed:
Update copy

fushar commented:
Thanks for the copy review. I updated the subtabs as follows:

<img width="450" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1525580/8078d011-1423-4a41-adf5-143eb940b30c">

1. Removed `logs` from the subtabs
2. `Web server` instead of `webserver` as this is what's used in the old Monitoring page 😂 

    <img width="570" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1525580/893b7690-a785-4e35-a264-47cde97358fa">

4. Keep `Log type` instead of `View`. If we go with `View` but we omit `logs`, it reads weird (e.g. we're not "viewing web server"; we're viewing the logs)
5. `Request Type` => `Request type` to match with the table heading below it.

    <img width="307" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1525580/125cc191-a716-44d9-bb28-5c79f9eed8fe">

This is a best-effort thinking and we can iterate later when needed~!

fushar commented:
> If I am in Logs > HTTP request logs of Site A, and then click on Site B from the collapsed Sites list, I land in Logs > PHP errors, but I expected to land on Logs > HTTP request logs of Site B. 

Yeah, I know. This is quite hard to implement as this needs a quite fundamental change to the architecture of how routing for the preview pane works 😭 let's create a new issue for this.

fushar commented:
Added this follow-up issue regarding persisting log type:
- https://github.com/Automattic/dotcom-forge/issues/7543
